src/methods/clipping/difference.jl

```diff
@@ -0,0 +1,292 @@
+                    # Add cartesian coordinates from inter_list
+                    push!(pt_list, (intr_list[current.idx][1], intr_list[current.idx][2]))
+                    
+                    # Keep track of processed intersection points
+                    if (current != a_list[starting_pt] && current != b_list[a_list[starting_pt].neighbor])
+                        processed_pts = processed_pts + 1
+                        tracker[current.idx] = typemax(Int)
+                    end
+                    
+                else
+                    # Add cartesian coordinates from "list", which should point to either a_list or b_list
+                    push!(pt_list, (list_edges[current.idx][1][1], list_edges[current.idx][1][2]))
+                end
+
+                current_node_not_intersection = !current.inter
+            end
+            
+            # Break once get back to starting point
+            current_node_not_starting = (current != a_list[starting_pt] && current != b_list[a_list[starting_pt].neighbor])
+
+            # Switch to neighbor list
+            if list == a_list
+                list = b_list
+                list_edges = edges_b
+            else
+                list = a_list
+                list_edges = edges_a
+            end
+            idx = current.neighbor
+            current = list[idx]
+        end
+
+        push!(return_polys, pt_list)
+    end
+
+    # If at this point return_polys contains multiple polygons,
+    # those polygons are all separate polygons. They cannot be holes
+    # of each other. This is because we took the difference of two polygon
+    # exteriors. If at this point return_polys is empty, it is possible
+    # that poly_b is entirely contained in poly_a and the list that this function
+    # returns will have length greater than 1 but only represent one polygon.
+    diff_polygons = length(return_polys) > 1
+
+    # Check if one polygon totally within other
+    if isempty(return_polys)
+        list_b = []
+        for i in eachindex(edges_b)
+            push!(list_b, edges_b[i][1])
+        end
+        push!(list_b, edges_b[1][1])
+
+        list_a = []
+        for i in eachindex(edges_a)
+            push!(list_a, edges_a[i][1])
+        end
+        push!(list_a, edges_a[1][1])
+
+        if point_in_polygon(edges_a[1][1], poly_b)[1]
+            return return_polys, diff_polygons
+        elseif point_in_polygon(edges_b[1][1], poly_a)[1]
+            push!(return_polys, list_a)
+            push!(return_polys, list_b)
+            return return_polys, diff_polygons
+
+        else
+            # equivalent of push!(return_polys, list_a) since return_polys empty
+            # This is the case where the two polygons don't intersect and are not
+            # contained in one another. Thus, we are returning the original poly_a
+            return [list_a], false
+        end
+    end
+
+    return return_polys, diff_polygons
+end
+
+"""
```

Comment:
Let's put this guy at the top above the helper functions so that it is easy to find!